src/hooks/libp2p/useBootLibp2pNode.ts

```diff
@@ -0,0 +1,62 @@
+import { nodeConfig } from 'lib/config/node_config';
+import { createLibp2p, Libp2p } from 'libp2p';
+import { useCallback, useEffect } from 'react';
+import { setLibp2p } from 'store/app/actions';
+import { useDispatch, useSelector } from 'store/index';
+import { log } from '../../lib/utils/logger';
+import { useLibp2p } from './useLibp2p';
+
+/**
+ * This hook is responsible for initializing a libp2p node
+ * The browser uses this node to communicate with archaeologists
+ * who themselves are libp2p nodes
+ *
+ * Boot Sequence:
+ * 1. Initialize node with a pre-defined config
+ * 2. Start the node
+ * 3. Add event listeners to the node for peer discovery and connection
+ */
+
+export function useBootLibp2pNode() {
+  const dispatch = useDispatch();
+  const globalLibp2pNode = useSelector(s => s.appState.libp2pNode);
+  const { handlePublicKeyStream, onPeerConnect, onPeerDisconnect, onPeerDiscovery } = useLibp2p();
+
+  const createAndSetNode = useCallback(async (): Promise<Libp2p> => {
+    const newLibp2pNode = await createLibp2p(nodeConfig);
+
+    // set global libp2p node for later use
+    dispatch(setLibp2p(newLibp2pNode));
+
+    return newLibp2pNode;
+  }, [dispatch]);
+
+  const addNodeEventListeners = useCallback(
+    (libp2pNode: Libp2p): void => {
+      libp2pNode.addEventListener('peer:discovery', onPeerDiscovery);
+      libp2pNode.connectionManager.addEventListener('peer:connect', onPeerConnect);
+      libp2pNode.connectionManager.addEventListener('peer:disconnect', onPeerDisconnect);
+    },
+    [onPeerDiscovery, onPeerConnect, onPeerDisconnect]
+  );
+
+  useEffect(() => {
+    (async () => {
+      try {
+        if (!globalLibp2pNode) {
+          const newLibp2pNode = await createAndSetNode();
+          await newLibp2pNode.start();
+
+          log(`Browser node starting with peerID: ${newLibp2pNode.peerId.toString()}`);
+
+          addNodeEventListeners(newLibp2pNode);
+          // TODO: remove event listeners
+          // Need to find correct place to call this without disrupting the event listeners
+          // Was previously removing on unmount of the useLibp2p hook, but that was not working
```

Comment:
There is still potential for these listeners to be added multiple times in the case where `globalLibp2pNode` is undefined so `createAndSetNode()` triggers but this useEffect fires again (maybe by another component) before the `libp2pNode` has a chance to be set in the store.  

That being said, consequences are not severe and this hook is only used once, so it should be fine to leave it.

Reply:
One untested possible solution is to add a second useEffect just for adding the listeners, with `libp2pNode` in the dependency list. 
```
useEffect(() => {
  if (libp2pNode) {
    addNodeEventListeners(newLibp2pNode);
  }
  return () => {
    // destroy listeners
  }
}, [libp2pNode])
```
This way if `libp2pNode` is undefined the listeners will not be added, but as soon as it is defined in the store, then the listeners will be added.

Haven't tested this so it may not work.

Reply:
oh wait, no nvm i see what you are saying -- b/c `dispatch` doesn't run sync, this could be run again. I should test this scenario -- what happens when listeners are added multiple times. Will report back with test results

Reply:
It's possible nothing bad happens when duplicate listeners are created... I'm curious if it would show duplicate archs.

Reply:
I think you are right... Any previous listeners are probably being destroyed if the old old node is replaced with a new one.